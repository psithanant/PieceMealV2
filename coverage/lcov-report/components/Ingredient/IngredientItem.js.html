<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Ingredient/IngredientItem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Ingredient</a> IngredientItem.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import {
    Icon,
    Card,
    CardItem,
    Thumbnail,
    Text,
    Right } from 'native-base';
// import RecipeModal from '../Recipe/RecipeModal';
import iceCream from '../../assets/food/ice-cream.jpg';
import bruscetta from '../../assets/food/Bruscetta.jpg';
import pepperBeef from '../../assets/food/pepperBeef.jpg';
&nbsp;
const images = [iceCream, bruscetta, pepperBeef];
&nbsp;
export default class IngredientItem extends Component {
&nbsp;
  // state = {
  //     modalVisible: false,
  //     name: '',
  //     description: '',
  //     notes: ''
  // }
&nbsp;
  render() {
      const rowData = this.props.rowData;
      return (
          &lt;Card&gt;
            &lt;TouchableOpacity onPress={this.props.onPress}&gt;
              &lt;CardItem&gt;
                  &lt;Thumbnail source={images[rowData.id % images.length]} /&gt;
                  &lt;Text style={{ alignSelf: 'center', marginLeft: 25 }}&gt;{ rowData.name }&lt;/Text&gt;
                  &lt;Right&gt;
                  &lt;Icon name="arrow-forward" /&gt;
                  &lt;/Right&gt;
              &lt;/CardItem&gt;
            &lt;/TouchableOpacity&gt;
          &lt;/Card&gt;
      );
  }
}
&nbsp;
// const mapStateToProps = (state) =&gt; {
//     return {
//     };
// };
//
// export default connect(mapStateToProps)(IngredientItem);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Jun 16 2017 14:33:43 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
