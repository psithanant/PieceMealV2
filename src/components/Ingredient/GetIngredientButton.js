import React from 'react';
import { connect } from 'react-redux';
import { Button, Text } from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { getIngredients } from '../../actions/index';

export const GetIngredientsButton = props => (
  <Button onPress={props.getAllIngredients} style={styles.buttonStyle} iconLeft>
    <Icon name='food-variant' size={35} />
    <Text>See All Ingredients</Text>
  </Button>
);

const styles = {
  buttonStyle: {
    marginLeft: 5,
    marginRight: 5,
  }
};

export default connect(null, { getIngredients })(GetIngredientsButton);
