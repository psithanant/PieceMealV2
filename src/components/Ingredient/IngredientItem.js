import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import {
    Icon,
    Card,
    CardItem,
    Thumbnail,
    Text,
    Right } from 'native-base';
import DEFAULT_IMAGE from '../../assets/food/ice-cream.jpg';

export default class IngredientItem extends Component {
  render() {
    const rowData = this.props.rowData;
    let name = rowData.name || '';
    name = name.split(' ').map((str) => {
      return str.substring(0, 1).toUpperCase() + str.substring(1);
    }).join(' ');
    let image;
    if (rowData.image_url) {
      image = { uri: rowData.image_url };
    } else {
      image = DEFAULT_IMAGE;
    }
    return (
        <Card>
          <TouchableOpacity onPress={this.props.onPress}>
            <CardItem>
              <Thumbnail source={image} />
                <Text style={styles}>{name}</Text>
                <Right>
                  <Icon name="arrow-forward" style={{ color: '#68BAA7' }} />
                </Right>
            </CardItem>
          </TouchableOpacity>
        </Card>
    );
  }
}

const styles = {
  alignSelf: 'center',
  marginLeft: 25,
  fontFamily: 'Futura',
  fontSize: 16,
  color: '#373737'
};
