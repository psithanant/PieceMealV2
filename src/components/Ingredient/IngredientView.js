import React, { Component } from 'react';
import { Image, View } from 'react-native';
import { connect } from 'react-redux';
import Toast from 'react-native-simple-toast';
import { Badge, Content, Card, CardItem, Text, Body, Button } from 'native-base';
import { deleteIngredient } from '../../actions';
import DEFAULT_IMAGE from '../../assets/food/ice-cream.jpg';

export class IngredientView extends Component {

    navigateButton = (text) => {
      return (
        <Button
            style={styles.buttonStyle}
            small
            warning
            onPress={() => {
             this.props.navigation.navigate('EditIngredient', this.props.ingredient);
         }}
        >
          <Text style={styles.textStyle}>{text}</Text>
        </Button>
      );
    }

    render() {
      console.log(this.props.ingredient);
      const { ingredient, user } = this.props;
      const ingredientTags = ingredient.tags;
      const ingredientAlts = ingredient.alternatives;
      let alternativeNames = [];
      let name = ingredient.name || '';
      let image;

      alternativeNames = ingredientAlts.map((ingredientAlt) => {
        return ingredientAlt.name;
      });

      name = name.split(' ').map((str) => {
        return str.substring(0, 1).toUpperCase() + str.substring(1);
        }).join(' ');

      if (ingredient.image_url) {
        image = { uri: ingredient.image_url };
      } else {
        image = DEFAULT_IMAGE;
      }
        return (
            <Content>
                <Card style={{ flex: 1 }}>
                  <CardItem
                    header
                    style={{ alignSelf: 'center' }}
                  >
                    <Text style={styles.headerStyle}>{name}</Text>
                  </CardItem>
                  <CardItem>
                    <Body>
                      <Image style={styles.imageStyle} source={image} />
                        <Text
                          note
                          style={{
                            fontFamily: 'Futura',
                            marginLeft: 20,
                            marginRight: 15,
                            marginTop: 15,
                            marginBottom: 10,
                            alignSelf: 'center'
                          }}
                        >
                          {ingredient.description}
                        </Text>
                    </Body>
                  </CardItem>
                  {alternativeNames.length > 0 && <Text style={{ marginLeft: 15 }}>Alternatives:</Text>}
                  <CardItem>
                    {alternativeNames.map((alternative) => (
                      <Button small rounded style={{ backgroundColor: '#68BAA7', marginRight: 10 }}>
                        <Text style={styles.textStyle}>{alternative}</Text>
                      </Button>
                    ))}
                  </CardItem>
                  {ingredientTags.length > 0 && <Text style={{ marginTop: 5, marginLeft: 15 }}>Tags: </Text>}
                  <CardItem>
                      {/* <View style={styles.viewStyle}> */}

                      {ingredientTags.map((ingredientTag) => (
                        <Badge key={ingredientTags.indexOf(ingredientTag)} style={{ backgroundColor: '#556270', marginRight: 10 }}>
                            <Text style={styles.textStyle}>{ingredientTag}</Text>
                        </Badge>))}
                      {/* </View> */}
                    </CardItem>
                    <CardItem style={{ marginBottom: 30, marginTop: 30 }} footer>
                      {this.props.user && this.navigateButton('Edit')}
                      {this.props.user &&
                        <Button
                        style={styles.buttonStyle}
                        danger
                        small
                        onPress={() => {
                         this.props.deleteIngredient(ingredient.id, user.token)
                           .then(res => {
                             if (res.value.status !== 204) {
                               return Toast.show('Bad Request');
                             }
                             return Toast.show('Item Deleted!');
                           })
                           .then(() => {
                             return this.props.navigation.navigate('ingredients');
                           });
                         }}
                        >
                          <Text style={styles.textStyle}>Delete</Text>
                      </Button>}
                    </CardItem>
                </Card>
            </Content>
        )
    }
}

const mapStateToProps = (state) => {
    return {
      // list: state.ingredients.ingredients,
      // isFetching: state.ingredients.isFetching,
      user: state.loginReducer.user
    };
};

export default connect(mapStateToProps, { deleteIngredient })(IngredientView);

const styles = {
  badgeStyle: {
    backgroundColor: 'black',
    // marginRight: 5,
    // flex: 1,
    // flexDirection: 'row',
    // alignItems: 'flex-start',
    // justifyContent: 'center',
    // width: -5
  },
  textStyle: {
    color: 'white',
    textAlign: 'center'
  },
  imageStyle: {
    resizeMode: 'contain',
    width: 340,
    height: 200
  },
  headerStyle: {
    fontFamily: 'Futura',
    fontSize: 24,
    marginTop: 20,
    paddingTop: 15
  },
  buttonStyle: {
    flex: 1,
    marginRight: 5,
    marginLeft: 5,
    alignSelf: 'center',
  },
  viewStyle: {
    flexDirection: 'row',
    height: 80,
    padding: 20,
    marginTop: 0,
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    alignItems: 'stretch',
    flex: 2
  }
};
