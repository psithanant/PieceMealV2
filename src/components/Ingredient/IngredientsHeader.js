import React, { Component } from 'react';
import { Header, Button, Icon, Title, Body, Right, Left } from 'native-base';

class IngredientHeader extends Component {

  render() {
    return (
      <Header>
        <Left />
          <Body>
              <Title>Ingredients</Title>
          </Body>
          <Right>
            <Button
              onPress={() => this.props.navigationHero.navigation.navigate('Home')}
              transparent
            >
              <Icon name="add" size={35} />
            </Button>
          </Right>
      </Header>
    );
  }
}

export default IngredientHeader;
