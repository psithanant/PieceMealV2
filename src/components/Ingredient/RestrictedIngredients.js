import React, { Component } from 'react';
import { ListView, TouchableWithoutFeedback, View } from 'react-native';
import { Button, ListItem, Text, Icon } from 'native-base';

export default class CheckBoxExample extends Component {

  loadDataSource() {
      const ds = new ListView.DataSource({
          rowHasChanged: (r1, r2) => r1 !== r2
      });

      this.dataSource = ds.cloneWithRows(this.props.list);
  }

  render() {
    this.loadDataSource();
    return (
      <TouchableWithoutFeedback
        onPress={() => this.onPress()}
      >
        <View>
          <ListView
            dataSource={this.dataSource}
            enableEmptySections
            renderRow={
              (rowData) =>
                <ListItem style={styles.containerStyle}>
                  <Button primary small>
                    <Icon name='beer' />
                  </Button>
                  <Text>{rowData.name}</Text>
                </ListItem>}
          />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = {
  containerStyle: {
    borderBottomWidth: 1,
    borderColor: '#ddd'
  },
  expandedContainerStyle: {
    backgroundColor: '#e5e5e5',
    borderBottomWidth: 1,
    borderColor: '#ccc'
  },
  nameStyle: {
    fontSize: 18,
    paddingLeft: 15
  }
};
