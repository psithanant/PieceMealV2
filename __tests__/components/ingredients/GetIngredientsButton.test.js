import 'react-native';
import React from 'react';
import { GetIngredientsButton } from '../../../src/components/Ingredient/GetIngredientButton';
import renderer from 'react-test-renderer';

describe('tests the GetIngredientsButton component', () => {
  it('renders getIngredients component properly', () => {
    expect(renderer.create(
      <GetIngredientsButton
        getIngredients={jest.fn}
      />
    )).toMatchSnapshot();
  });
});
