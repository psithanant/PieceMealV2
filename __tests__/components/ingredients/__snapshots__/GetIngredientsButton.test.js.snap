// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the GetIngredientsButton component renders getIngredients component properly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#007aff",
      "borderColor": "#007aff",
      "borderRadius": 5,
      "borderWidth": null,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "marginLeft": 5,
      "marginRight": 5,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 6,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 35,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï‰œ
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
      }
    }
    virtual={undefined}
  >
    See All Ingredients
  </Text>
</View>
`;
